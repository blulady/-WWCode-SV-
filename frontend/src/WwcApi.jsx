import axios from "axios";

const BASE_URL =
  process.env.REACT_APP_API_URL ||
  "https://wwcode-chtools-api-dev.herokuapp.com/api";

const getConfig = () => {
  let { access } = JSON.parse(sessionStorage.getItem("token"));
  return {
    Authorization: `Bearer ${access}`,
    "Content-Type": "application/json",
  };
};

axios.interceptors.response.use(
  (response) => {
    return response;
  },
  function (error) {
    
    const {detail,code, error: err} = error.response.data
    console.log(err);
    console.log('Interceptor error:',detail,':',code)
     
    const originalRequest = error.config;
    
    if (error.response.status === 401 && !originalRequest._retry && code==='token_not_valid'){
      // Access Token expired, get new token using the existing Refresh token
      if (sessionStorage.getItem("token") && detail==='Given token not valid for any token type'){
          originalRequest._retry = true;
          const { refresh } = JSON.parse(sessionStorage.getItem("token"));
          return axios
            .post(`${BASE_URL}/login/refresh`, { refresh: refresh })
            .then((res) => {
              if (res.status === 200) {
                console.log(res);
                let { access, refresh } = res.data;
                let token = { access, refresh };
                sessionStorage.setItem("token", JSON.stringify(token));
                console.log("Access token has been refreshed");
                originalRequest.headers = getConfig();
                return axios(originalRequest);
              } 
            });   
          } 
          // Refresh Token expired. clean up the tokens stored in sessionStorage
          if (detail==='Token is invalid or expired'){
            console.log("Refresh token expired, cleanup tokens and navigate to Login page")
            sessionStorage.removeItem("token");
            sessionStorage.removeItem("user");
            return window.location.href = `${BASE_URL}/login`;
          }   
    }
    return Promise.reject(error);
  }
);

class WwcApi {
  static async login(data) {
    return await axios.post(`${BASE_URL}/login`, data, {
      headers: { "content-type": "application/json" },
    });
  }

  static async sendResetEmail(email) {
    return await axios.post(
      `${BASE_URL}/user/reset_password/request/`,
      { email: email },
      {
        headers: { "content-type": "application/json" },
      }
    );
  }

  static async resetPassword(data){
    return await axios.patch(`${BASE_URL}/user/reset_password/confirm/`, data, {
      headers: { "Content-Type": "application/json" }});
  }

  static async logout() {
    return await axios.post(
      `${BASE_URL}/logout/`,
      {},
      {
        headers: getConfig(),
      }
    );
  }

  static async getMembers(orderBy, search) {
    let url = `${BASE_URL}/users/`;
    if (orderBy) {
      url += "?ordering=" + orderBy;
    }
    if (search) {
      url += (orderBy ? "&" : "?") + "search=" + search
    }
    let res = await axios.get(url, {
      headers: getConfig(),
    });
    return res.data;
  }

  static async getMember(userId) {
    let res = await axios.get(`${BASE_URL}/user/${userId}`, {
      headers: getConfig(),
    });
    return res.data;
  }

  static async createMember(data) {
    return await axios.post(`${BASE_URL}/user/create/`, data, {
      headers: getConfig(),
    });
  }

  static async register(data) {
    return await axios.post(`${BASE_URL}/user/activate/`, data);
  }

  static async getTeams() {
    let res = await axios.get(`${BASE_URL}/teams/`,{
      headers: getConfig(), 
    });
    return res.data;
  }

  static async editMember(data, userId) {
    return await axios.post(`${BASE_URL}/user/edit/${userId}`, data,{
      headers: getConfig(),
    });
  }

  static async getVolunteerResources(slug) {
    let res = await axios.get(`${BASE_URL}/resources/${slug}`, {
      headers: getConfig(),
    });
    return res.data;
  }

  static async changeMemberStatus(userId,data){
    return await axios.post(`${BASE_URL}/user/edit/${userId}/status`, data, {
      headers: getConfig(),
    });
  }

  static async deleteMemberRole(userId,data){
    return await axios.delete(`${BASE_URL}/user/edit/${userId}/role`, data, {
      headers: getConfig(),
    });
  }
}
export default WwcApi;
