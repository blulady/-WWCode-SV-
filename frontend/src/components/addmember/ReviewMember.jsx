import React, { useState } from "react";
import { useLocation } from "react-router-dom";
import ContainerWithNav from "../layout/ContainerWithNav";
import EditIcon from "../../images/edit_24px.png";
import EmailInput from "./EmailInput";
import RoleRadioInput from "./RoleRadioField";
import InputLabel from "./InputLabel";
import TextAreaInput from "./TextAreaInput";
import ConfirmationModal from "./ConfirmationModal";
import { useNavigate } from "react-router-dom";
import WwcApi from "../../WwcApi";
import MessageBox from "../messagebox/MessageBox";
import { ERROR_REQUEST_MESSAGE } from "../../Messages";
import BackToMemberPortal from "../layout/BackToMemberPortal";

function ReviewMember(props) {
  const navigate = useNavigate();
  const location = useLocation();
  const pageId = location.state.pageId;
  const pending = location.state.pending;
  const data = location.state.memberinfo;
  const roleinfo = location.state.roleinfo;
  const [errorOnRequest, setErrorOnRequest] = useState(false);

  const [member, setMember] = useState({
    Email: data.Email,
    Role: data.Role,
    Message: data.Message,
  });

  const [memberRole, setMemberRole] = useState({
    MemberRoleId: roleinfo.MemberRoleId,
    MemberRole: roleinfo.MemberRole,
    MemberRoleDescription: roleinfo.MemberRoleDescription,
  });

  const [edit, setEdit] = useState({
    EmailInput: false,
    RoleInput: false,
    MessageInput: false,
  });
  const MemberRoleArray = [
    { id: "1", role: "Volunteer", desc: "Limited access to areas of portal" },
    {
      id: "2",
      role: "Leader",
      desc: "Access to all areas excluding Director area",
    },
    { id: "3", role: "Director", desc: "Access to all areas of portal" },
  ];
  const [showModal, setShowModal] = useState(false);
  const handleRadioClick = (id) => {
    MemberRoleArray.forEach((item) => {
      if (item.id === id) {
        setMemberRole({
          MemberRoleId: item.id,
          MemberRole: item.role,
          MemberRoleDescription: item.desc,
        });
        setShowModal(true);
      }
    });
  };

  const handleChange = (event) => {
    setMember({ ...member, [event.target.name]: event.target.value });
  };

  const onClickSendInvite = async (event) => {
    if (!edit.EmailInput && !edit.RoleInput && !edit.MessageInput) {
      //make the axios call to save the new member
      //if successful display the successModal
      event.preventDefault();
      const memberInfo = {
        email: member.Email,
        role: member.Role.toUpperCase(),
        message: member.Message,
      };
      try {
        setErrorOnRequest(false);
        const results = await WwcApi.addInvitee(memberInfo);
        navigate("/member/add",
          { state: { fromReview: true, pageId, pending }});
      } catch (error) {
        setErrorOnRequest(true);
        console.log(error + ':\n'+ JSON.stringify(error.response.data));
      }
    }
  };
  const handleSubmit = (event) => {
    event.preventDefault();
    if (edit.EmailInput) {
      setEdit({ EmailInput: false });
    }
    if (edit.RoleInput) {
      setEdit({ RolelInput: false });
    }
    if (edit.MessageInput) {
      setEdit({ MessageInput: false });
    }
  };
  const onClickMessageEdit = (event) => {
    if (!edit.EmailInput && !edit.RoleInput) {
      setEdit({ MessageInput: true });
    }
  };

  const onClickRoleEdit = (event) => {
    if (!edit.MessageInput && !edit.EmailInput) {
      setEdit({ RoleInput: true });
    }
  };

  const onClickEmailEdit = (event) => {
    if (!edit.MessageInput && !edit.RoleInput) {
      setEdit({ EmailInput: true });
    }
  };

  return (
    <ContainerWithNav>
      <main>
        <div className='container'>
          <div className='button-inside-div'>
            <button
              className='chapter-member-btn'
              onClick={() => {
                navigate("/chapter/members");
              }}
            >
              Chapter Members
            </button>
          </div>
          <div className='form-div'>
          <BackToMemberPortal pageId={pageId} pending={pending} />
            <div className='row justify-content-center form-div-spacing'>
            <div className={errorOnRequest ? "show padded" : "hide"}>
              <MessageBox type="Error" title={"Sorry!"} message={ERROR_REQUEST_MESSAGE}></MessageBox>
            </div>
              <div>
                <div className='header'>Review Member to Be</div>
                <div className='header'>Added</div>
                <p className='subheader'> *Mandatory Fields</p>
                <form onSubmit={handleSubmit}>
                  <div className='row justify-content-center reviewform-border'>
                    <section className='review-email-div'>
                      <InputLabel labelName='email' text='Email *' />
                      <EmailInput
                        name='Email'
                        value={member.Email}
                        emailvalue={member.Email}
                        className={
                          edit.EmailInput
                            ? "form-control email-input show"
                            : "form-control email-input hide"
                        }
                        pclass={edit.EmailInput ? "hide" : "show"}
                        editclass={
                          edit.EmailInput
                            ? "hide"
                            : "show email-editicon-transform"
                        }
                        buttonclass={
                          edit.EmailInput
                            ? "show save-btn responsive-save-btn "
                            : "hide save-btn"
                        }
                        editiconclick={onClickEmailEdit}
                        onChange={handleChange}
                      />
                    </section>
                    <section className=' review-role-div'>
                      <InputLabel labelName='role' text='Role *' />
                      <div className='div-inline responsive-div-block'>
                        <div
                          className={
                            edit.RoleInput ? "hide" : "show responsive-role-div"
                          }
                        >
                          <div className='review-member-role'>
                            {memberRole.MemberRole}
                          </div>
                          <div className='review-role-desc'>
                            {memberRole.MemberRoleDescription}
                          </div>
                        </div>
                        <div className={edit.RoleInput ? "show" : "hide"}>
                          {MemberRoleArray.map((item) => (
                            <div
                              key={item.id}
                              className={
                                memberRole.MemberRoleId === item.id
                                  ? "form-check radio-btn-selected"
                                  : "form-check radio-input"
                              }
                            >
                              <RoleRadioInput
                                checked={memberRole.MemberRoleId === item.id}
                                id={item.id}
                                name='Role'
                                value={item.id}
                                pclass={edit.RoleInput ? "hide" : "show"}
                                datatarget='#confirmModal'
                                onClick={() => handleRadioClick(item.id)}
                                onChange={handleChange}
                                roletext={item.role}
                                roledesc={item.desc}
                              />
                            </div>
                          ))}
                        </div>
                        <div className='spacer' />

                        <button
                          type='submit'
                          className={
                            edit.RoleInput
                              ? "show save-btn responsive-save-btn"
                              : "hide"
                          }
                        >
                          Save
                        </button>
                        <div className='edit-icon'>
                          <img
                            className={
                              edit.RoleInput
                                ? "hide"
                                : "show role-editicon-transform"
                            }
                            src={EditIcon}
                            style={{ width: "auto", height: "auto" }}
                            alt='Edit'
                            onClick={onClickRoleEdit}
                          />
                        </div>
                      </div>
                    </section>
                    <section className='review-message-div'>
                      <InputLabel
                        labelName='message'
                        text='Message (optional)'
                      />
                      <TextAreaInput
                        name='Message'
                        memberclass={edit.MessageInput ? "hide" : "show "}
                        message={member.Message}
                        className={
                          edit.MessageInput
                            ? "form-control message-textarea show"
                            : "hide"
                        }
                        onChange={handleChange}
                        value={member.Message}
                        countervalue={
                          member.Message
                            ? member.Message.length + "/2000 char"
                            : null
                        }
                        editclass={
                          edit.MessageInput
                            ? "hide"
                            : "show message-editicon-transform"
                        }
                        editiconclick={onClickMessageEdit}
                        counterclass={
                          edit.MessageInput ? "show message-counter" : "hide"
                        }
                        buttonclass={
                          edit.MessageInput
                            ? "show save-btn responsive-save-btn "
                            : "hide"
                        }
                      />
                    </section>
                  </div>
                  <div className='form-group d-flex justify-content-center'>
                    <input
                      type='button'
                      className='d-flex justify-content-center sendinvite-btn'
                      onClick={onClickSendInvite}
                      value='Send Invite'
                    />
                  </div>
                </form>
                {/* Role Confirmation Modal code starts */}
                <ConfirmationModal
                  style={{ display: showModal ? "show" : "hide" }}
                  onClick={() => {
                    setShowModal(false);
                  }}
                  memberrole={memberRole.MemberRole}
                  memberdesc={memberRole.MemberRoleDescription}
                />
                {/* Role Confirmation Modal code ends */}
              </div>
            </div>
          </div>
        </div>
      </main>
    </ContainerWithNav>
  );
}
export default ReviewMember;
