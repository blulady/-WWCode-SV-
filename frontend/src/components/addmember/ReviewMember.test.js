import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ReviewMember from './ReviewMember';
import WwcApi from '../../WwcApi';

const mockHistoryPush = jest.fn();
let mockMemberInfo = {
    Email: '',
    Role: '',
    Message: ''
}
let mockRoleInfo = {
    MemberRoleId: '',
    MemberRole: '',
    MemberRoleDescription: ''
};

jest.mock('react-router-dom', () => {
    const ActualReactRouterDom = require.requireActual('react-router-dom');
    return {
        ...ActualReactRouterDom,
        useHistory: () => ({
        push: mockHistoryPush,
        }),
        useLocation: () => ({
            state: {
                memberinfo: mockMemberInfo,
                roleinfo: mockRoleInfo
            }
        })
    }
});

jest.mock('react', () => {
    const ActualReact = require.requireActual('react');
    return {
        ...ActualReact,
        useContext: () => ({
            userInfo: {
                email: 'director@example.com',
                first_name: 'John',
                id: 1,
                last_name: 'Smith',
                role: 'DIRECTOR'
            }
        }),
    };
});

describe('ReviewMember', () => {
    beforeEach(() => {
        mockMemberInfo = {
            Email: 'test@example.com',
            Role: 'Director',
            Message: 'hello world!'
        };
        mockRoleInfo = {
            MemberRoleId: '3',
            MemberRole: 'Director',
            MemberRoleDescription: 'Access to all areas of portal'
        };
    });

    test('it renders without crashing', () => {
        const { container } = render(<ReviewMember />);
        
        expect(container).toMatchSnapshot();
    });

    test('it displays information correctly', () => {
        const { container, getByTestId, getAllByText } = render(<ReviewMember />);
        const email = container.querySelector('.email-input');
        const role = container.querySelector('.review-member-role');
        const roleDesc = container.querySelector('.review-role-desc');
        const message = getByTestId('message');

        expect(email.value).toBe('test@example.com');
        expect(role).toHaveTextContent('Director');
        expect(roleDesc).toHaveTextContent('Access to all areas of portal');
        expect(message.value).toBe('hello world!');
        expect(getAllByText('Save')[0]).not.toHaveClass('responsive-save-btn');
        expect(getAllByText('Save')[1]).not.toHaveClass('responsive-save-btn');
        expect(getAllByText('Save')[2]).not.toHaveClass('responsive-save-btn');
    });

    test('it allows email to be editable', () => {
        const { container, getAllByText } = render(<ReviewMember />);
        const edit = container.querySelector('.email-editicon-transform');

        fireEvent.click(edit);

        expect(getAllByText('Save')[0]).toHaveClass('responsive-save-btn');
    });

    test('it allows role to be editable', () => {
        const { container, getAllByText } = render(<ReviewMember />);
        const edit = container.querySelector('.role-editicon-transform');

        fireEvent.click(edit);

        expect(getAllByText('Save')[1]).toHaveClass('responsive-save-btn');
    });

    test('it allows message to be editable', () => {
        const { container, getAllByText } = render(<ReviewMember />);
        const edit = container.querySelector('.message-editicon-transform');

        fireEvent.click(edit);

        expect(getAllByText('Save')[2]).toHaveClass('responsive-save-btn');
    });

    test('it redirects on send invite click', async (done) => {
        const apiSpy = jest.spyOn(WwcApi, 'createMember').mockReturnValue(await Promise.resolve('success'));
        const { container } = render(<ReviewMember />);
        const sendInvite = container.querySelector('.sendinvite-btn');

        fireEvent.click(sendInvite);

        await expect(apiSpy).toHaveBeenCalledTimes(1);
        expect(mockHistoryPush).toHaveBeenCalledWith({
            pathname: '/member/add',
            state: {
                fromReview: true
            }
        });
        done();
    });

    test('it shows error', () => {
        function CustomException(message) {
            const error = new Error(message);
          
            error.response = {
                data: message
            }

            return error;
        }
          
        CustomException.prototype = Object.create(Error.prototype);
        const exception = new CustomException('createMember failed')
        const apiSpy = jest.spyOn(WwcApi, 'createMember').mockImplementation(() => {
            throw exception;
        });
        const alertSpy = jest.spyOn(window, 'alert').mockImplementation(() => {});
        const { container } = render(<ReviewMember />);
        const sendInvite = container.querySelector('.sendinvite-btn');

        fireEvent.click(sendInvite);

        expect(apiSpy).toHaveBeenCalled();
        expect(alertSpy).toHaveBeenCalledWith(exception + ':\n' + JSON.stringify(exception.response.data));
    });
});