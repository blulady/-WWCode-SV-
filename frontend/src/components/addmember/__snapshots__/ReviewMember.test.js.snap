// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewMember it renders without crashing 1`] = `
<div>
  <div
    class="container-fluid container-fluid"
  >
    <nav
      class="container-navbar"
    >
      <div
        class="container-inner"
      >
        <div
          class="navbar-brand navbar-wwc-logo"
        >
          <div
            class="logo-png"
          />
        </div>
        <div
          class="navbar-items "
        >
          <ul
            class="navbar-items-ul"
          >
            <li
              class="navbar-text navbar-items-wwc-title"
            >
              Chapter Tools
            </li>
            <li
              class="navbar-items-user-button"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="user-button"
                data-bs-toggle="dropdown"
                id="navbarDropdown"
                role="button"
              >
                <span>
                  J
                </span>
              </div>
              <div
                aria-labelledby="navbarDropdown"
                class="dropdown-menu dropdown-menu"
              >
                <p
                  class="dropdown-item signed-user dropdown-item"
                >
                  Signed-in as
                </p>
                <p
                  class="font-weight-bold dropdown-item signed-user dropdown-item"
                >
                  director@example.com
                </p>
                <button
                  class="dropdown-item dropdown-item"
                >
                  Your Profile
                </button>
                <hr
                  class="dropdown-divider dropdown-hr"
                />
                <button
                  class="dropdown-item item-logout dropdown-item"
                >
                  Log Out
                </button>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="container-body"
    >
      <main>
        <div
          class="container"
        >
          <div
            class="button-inside-div"
          >
            <button
              class="chapter-member-btn"
            >
              Chapter Members
            </button>
          </div>
          <div
            class="form-div"
          >
            <div>
              <button
                class="back-member-btn"
              >
                <img
                  alt="Back Button"
                  class="back-btn-img"
                  src="arrow_back_24px.png"
                />
                Back to 
                Test Page
              </button>
            </div>
            <div
              class="row justify-content-center form-div-spacing"
            >
              <div
                class="hide"
              >
                <div
                  class="error message-box d-flex align-items-center justify-content-center flex-column"
                  data-testid="message-box"
                >
                  <div
                    class="title"
                  >
                    Sorry!
                  </div>
                  <div
                    class="message"
                  >
                    We could not process your request. 
Please try again.
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="header"
                >
                  Review Member to Be
                </div>
                <div
                  class="header"
                >
                  Added
                </div>
                <p
                  class="subheader"
                >
                   *Mandatory Fields
                </p>
                <form>
                  <div
                    class="row justify-content-center reviewform-border"
                  >
                    <section
                      class="review-email-div"
                    >
                      <div>
                        <label
                          class="input-label"
                        >
                          Email *
                        </label>
                      </div>
                      <div
                        class="div-inline responsive-div-block"
                      >
                        <p
                          class="show"
                        >
                          test@example.com
                        </p>
                        <input
                          class="form-control email-input hide"
                          id="email"
                          name="Email"
                          required=""
                          type="email"
                          value="test@example.com"
                        />
                        <div
                          class="spacer"
                        />
                        <div
                          class="edit-icon"
                        >
                          <img
                            alt="Edit"
                            class="show email-editicon-transform"
                            src="edit_24px.png"
                            style="width: auto; height: auto;"
                          />
                        </div>
                        <button
                          class="hide save-btn"
                          type="submit"
                        >
                          Save
                        </button>
                      </div>
                    </section>
                    <section
                      class=" review-role-div"
                    >
                      <div>
                        <label
                          class="input-label"
                        >
                          Role *
                        </label>
                      </div>
                      <div
                        class="div-inline responsive-div-block"
                      >
                        <div
                          class="show responsive-role-div"
                        >
                          <div
                            class="review-member-role"
                          >
                            Director
                          </div>
                          <div
                            class="review-role-desc"
                          >
                            Access to all areas of portal
                          </div>
                        </div>
                        <div
                          class="hide"
                        >
                          <div
                            class="form-check radio-input"
                          >
                            <input
                              data-bs-target="#confirmModal"
                              data-bs-toggle="modal"
                              id="1"
                              name="Role"
                              type="radio"
                              value="1"
                            />
                            <div
                              class="member-role"
                            >
                              Volunteer
                            </div>
                            <div
                              class="role-desc"
                            >
                              Limited access to areas of portal
                            </div>
                          </div>
                          <div
                            class="form-check radio-input"
                          >
                            <input
                              data-bs-target="#confirmModal"
                              data-bs-toggle="modal"
                              id="2"
                              name="Role"
                              type="radio"
                              value="2"
                            />
                            <div
                              class="member-role"
                            >
                              Leader
                            </div>
                            <div
                              class="role-desc"
                            >
                              Access to all areas excluding Director area
                            </div>
                          </div>
                          <div
                            class="form-check radio-btn-selected"
                          >
                            <input
                              checked=""
                              data-bs-target="#confirmModal"
                              data-bs-toggle="modal"
                              id="3"
                              name="Role"
                              type="radio"
                              value="3"
                            />
                            <div
                              class="member-role"
                            >
                              Director
                            </div>
                            <div
                              class="role-desc"
                            >
                              Access to all areas of portal
                            </div>
                          </div>
                        </div>
                        <div
                          class="spacer"
                        />
                        <button
                          class="hide"
                          type="submit"
                        >
                          Save
                        </button>
                        <div
                          class="edit-icon"
                        >
                          <img
                            alt="Edit"
                            class="show role-editicon-transform"
                            src="edit_24px.png"
                            style="width: auto; height: auto;"
                          />
                        </div>
                      </div>
                    </section>
                    <section
                      class="review-message-div"
                    >
                      <div>
                        <label
                          class="input-label"
                        >
                          Message (optional)
                        </label>
                      </div>
                      <div
                        class="div-inline responsive-div-block"
                      >
                        <p
                          class="show "
                        >
                          hello world!
                        </p>
                        <div
                          style="display: block;"
                        >
                          <textarea
                            class="hide"
                            data-testid="message"
                            id="message"
                            maxlength="2000"
                            name="Message"
                            placeholder="Add an optional message to the registration email sent to the user (max 2000 char)"
                            rows="6"
                          >
                            hello world!
                          </textarea>
                          <div
                            class="hide"
                          >
                            12/2000 char
                          </div>
                        </div>
                        <div
                          class="spacer"
                        />
                        <div
                          class="edit-icon"
                        >
                          <img
                            alt="Edit"
                            class="show message-editicon-transform"
                            src="edit_24px.png"
                            style="width: auto; height: auto;"
                          />
                        </div>
                        <button
                          class="hide"
                          type="submit"
                        >
                          Save
                        </button>
                      </div>
                    </section>
                  </div>
                  <div
                    class="form-group d-flex justify-content-center"
                  >
                    <input
                      class="d-flex justify-content-center sendinvite-btn"
                      type="button"
                      value="Send Invite"
                    />
                  </div>
                </form>
                <div
                  class="modal"
                  id="confirmModal"
                  role="dialog"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog modal-dialog-centered"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-body modal-open-text"
                      >
                        *Selected role for this member is
                         
                        <span
                          class="confirm-message"
                        >
                          Director
                        </span>
                        , who will have
                         
                        <span
                          class="confirm-message"
                        >
                          Access to all areas of portal
                        </span>
                        <div
                          class="d-flex justify-content-center"
                        >
                          <button
                            class="modal-confirm-btn"
                            data-bs-dismiss="modal"
                            type="button"
                          >
                            Confirm
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
